src/pudl/metadata/classes.py

```diff
@@ -1632,3 +1633,45 @@ def to_sql(
                 check_values=check_values,
             )
         return metadata
+
+
+class CodeData(Base):
+    """
+    A collection of dictionaries representing codes for various PUDL fields.
+
+    Used to export to documentation.
+    """
+
+    code_list: List[dict]
+
+    @classmethod
+    def from_code_names(
+        cls, code_names: Iterable[str], csv_dir: String
+    ) -> "CodeData":
+        """
+        Construct a list of code dictionaries containing name, description, and file path to a CSV of the code label dataframe.
+
+        Args:
+            code_names: A list of Code PUDL identifiers, keys to entries in the CODE_METADATA dictionary.
+
+        """
+        code_list = []
+        for name in code_names:
+            if name in CODE_DESCRIPTIONS.keys():
+                description = CODE_DESCRIPTIONS[name]['description']
+                title_name = CODE_DESCRIPTIONS[name]['doc_name']
+            else:
+                description = None
+                title_name = name
+            csv_filepath = os.path.join(csv_dir, f"{name}.csv")
+            CODE_METADATA[name]["df"].to_csv(csv_filepath)
+            code_dict = {"title_name": title_name, "var_name": name, "description": description,
+                         "csv_filepath": csv_filepath}
+            code_list.append(code_dict)
+        return cls(code_list=code_list)
+
+    def to_rst(self, path: str) -> None:
```

Comment:
If this class contains a list of `Encoders` each of which knows has a `to_rst()` method then I think the `to_rst()` method here could just iterate over all of them calling their own `to_rst()` methods, and add whatever preamble / postscript / glue is required to turn them into a complete page.

Reply:
Could also have a `cleanup()` method that's called to remove the CSV files after the docs build has completed.